discord/message.py

```diff
@@ -837,6 +841,45 @@ def __init__(self, data: RoleSubscriptionDataPayload) -> None:
         self.is_renewal: bool = data['is_renewal']
 
 
+class GuildProductPurchase(NamedTuple):
+    """Represents a message's guild product that the user has purchased.
+
+    .. versionadded:: 2.5
+    """
+
+    listing_id: int
+    """The ID of the listing that the user has purchased."""
+    product_name: str
+    """The name of the product that the user has purchased."""
```

Comment:
We don't use NamedTuple in the public API if we can help it. Please just turn this into a regular class.